src/rustup-dist/src/dist.rs

```diff
@@ -448,6 +448,36 @@ impl<'a> Manifest<'a> {
     }
 }
 
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq)]
+pub enum Profile {
+    Minimal,
+    Default,
+    Complete,
+}
+
+impl Profile {
+    pub fn from_str(name: &str) -> Result<Self> {
+        match name {
+            "minimal" | "m" => Ok(Profile::Minimal),
+            "default" | "d" | "" => Ok(Profile::Default),
+            "complete" | "c" => Ok(Profile::Complete),
+            _ => Err(ErrorKind::InvalidProfile(name.to_owned()).into()),
+        }
+    }
+
+    pub fn names() -> &'static [&'static str] {
+        &["minimal", "default", "complete"]
+    }
+
+    // Non-required components that were required before profiles exist.
+    pub fn legacy_profile() -> Vec<String> {
```

Comment:
Is this only for upgrading to a profiled install?

Reply:
yes